// Code generated by mockery v2.44.2. DO NOT EDIT.

package mocks

import (
	context "context"

	mock "github.com/stretchr/testify/mock"
	entity "github.com/venture-technology/venture/internal/entity"
)

// IResponsibleRepository is an autogenerated mock type for the IResponsibleRepository type
type IResponsibleRepository struct {
	mock.Mock
}

// Auth provides a mock function with given fields: ctx, responsible
func (_m *IResponsibleRepository) Auth(ctx context.Context, responsible *entity.Responsible) (*entity.Responsible, error) {
	ret := _m.Called(ctx, responsible)

	if len(ret) == 0 {
		panic("no return value specified for Auth")
	}

	var r0 *entity.Responsible
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, *entity.Responsible) (*entity.Responsible, error)); ok {
		return rf(ctx, responsible)
	}
	if rf, ok := ret.Get(0).(func(context.Context, *entity.Responsible) *entity.Responsible); ok {
		r0 = rf(ctx, responsible)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*entity.Responsible)
		}
	}

	if rf, ok := ret.Get(1).(func(context.Context, *entity.Responsible) error); ok {
		r1 = rf(ctx, responsible)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// Create provides a mock function with given fields: ctx, responsible
func (_m *IResponsibleRepository) Create(ctx context.Context, responsible *entity.Responsible) error {
	ret := _m.Called(ctx, responsible)

	if len(ret) == 0 {
		panic("no return value specified for Create")
	}

	var r0 error
	if rf, ok := ret.Get(0).(func(context.Context, *entity.Responsible) error); ok {
		r0 = rf(ctx, responsible)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// Delete provides a mock function with given fields: ctx, cpf
func (_m *IResponsibleRepository) Delete(ctx context.Context, cpf *string) error {
	ret := _m.Called(ctx, cpf)

	if len(ret) == 0 {
		panic("no return value specified for Delete")
	}

	var r0 error
	if rf, ok := ret.Get(0).(func(context.Context, *string) error); ok {
		r0 = rf(ctx, cpf)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// FindByEmail provides a mock function with given fields: ctx, email
func (_m *IResponsibleRepository) FindByEmail(ctx context.Context, email *string) (*entity.Responsible, error) {
	ret := _m.Called(ctx, email)

	if len(ret) == 0 {
		panic("no return value specified for FindByEmail")
	}

	var r0 *entity.Responsible
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, *string) (*entity.Responsible, error)); ok {
		return rf(ctx, email)
	}
	if rf, ok := ret.Get(0).(func(context.Context, *string) *entity.Responsible); ok {
		r0 = rf(ctx, email)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*entity.Responsible)
		}
	}

	if rf, ok := ret.Get(1).(func(context.Context, *string) error); ok {
		r1 = rf(ctx, email)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// Get provides a mock function with given fields: ctx, cpf
func (_m *IResponsibleRepository) Get(ctx context.Context, cpf *string) (*entity.Responsible, error) {
	ret := _m.Called(ctx, cpf)

	if len(ret) == 0 {
		panic("no return value specified for Get")
	}

	var r0 *entity.Responsible
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, *string) (*entity.Responsible, error)); ok {
		return rf(ctx, cpf)
	}
	if rf, ok := ret.Get(0).(func(context.Context, *string) *entity.Responsible); ok {
		r0 = rf(ctx, cpf)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*entity.Responsible)
		}
	}

	if rf, ok := ret.Get(1).(func(context.Context, *string) error); ok {
		r1 = rf(ctx, cpf)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// SaveCard provides a mock function with given fields: ctx, cpf, cardToken, paymentMethodId
func (_m *IResponsibleRepository) SaveCard(ctx context.Context, cpf *string, cardToken *string, paymentMethodId *string) error {
	ret := _m.Called(ctx, cpf, cardToken, paymentMethodId)

	if len(ret) == 0 {
		panic("no return value specified for SaveCard")
	}

	var r0 error
	if rf, ok := ret.Get(0).(func(context.Context, *string, *string, *string) error); ok {
		r0 = rf(ctx, cpf, cardToken, paymentMethodId)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// Update provides a mock function with given fields: ctx, currentResponsible, responsible
func (_m *IResponsibleRepository) Update(ctx context.Context, currentResponsible *entity.Responsible, responsible *entity.Responsible) error {
	ret := _m.Called(ctx, currentResponsible, responsible)

	if len(ret) == 0 {
		panic("no return value specified for Update")
	}

	var r0 error
	if rf, ok := ret.Get(0).(func(context.Context, *entity.Responsible, *entity.Responsible) error); ok {
		r0 = rf(ctx, currentResponsible, responsible)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// NewIResponsibleRepository creates a new instance of IResponsibleRepository. It also registers a testing interface on the mock and a cleanup function to assert the mocks expectations.
// The first argument is typically a *testing.T value.
func NewIResponsibleRepository(t interface {
	mock.TestingT
	Cleanup(func())
}) *IResponsibleRepository {
	mock := &IResponsibleRepository{}
	mock.Mock.Test(t)

	t.Cleanup(func() { mock.AssertExpectations(t) })

	return mock
}
